### PR TITLE
docs/HowToUseLWIP: Add an explanation how to control flags and pbuf

### PR DESCRIPTION
**Pull request by SeonghoByeon**
- Add an explanation how to control flags depending on driver's status, and
how to manage pbuf for input/output.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [9215e4e68733f702efe8575801b806a451663e1d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9215e4e68733f702efe8575801b806a451663e1d] - Code Rule Check (C++) OK.

File: docs/HowToUseLWIP.md [hunk 1]
```diff
@@ -103,9 +103,49 @@ Please note that **netif_is_up()** returns whether the netif is up or not.
...
+or only NETIF_LWIP_LINK_UP is set to 0 without changing NETIF_LWIP_UP, depending on the device driver structure.
+
+#### Network buffer (struct pbuf *p)
+1. INPUT: To pass a packet up the TCP/IP stack by the device driver, **netif->input** registered by netif_add() API should be called with pbuf.
+To allocate pbuf and manage it in the driver level, please refer to APIs defined in pbuf.h (LwIP pbuf APIs).
+In general, **tcpip_input()** defined in tcpip.c is used as below.
+```
+err_t tcpip_input(struct pbuf *p, struct netif *netif_in);
+```
+NOTE: Even if pbuf allocation fails due to memory pull overflow from LwIP, the driver should not discard the packet rashly.
```
Comment (davidfather) [inline docs/HowToUseLWIP.md]: Do you mean memory pool here?

Comment (SeonghoByeon) [reply docs/HowToUseLWIP.md]: That's right. It's a typo. Let me modify it. Thanks!

**Comment by seinfra** [comment]
Target : [9a7ec12dc9137617b01a48e5905ff586e239ed2f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9a7ec12dc9137617b01a48e5905ff586e239ed2f] - Code Rule Check (C++) OK.

**Pull request merged**